polars/dataframe.ts

```diff
@@ -1755,6 +1801,66 @@ export const _DataFrame = (_df: any): DataFrame => {
     set columns(names) {
       _df.columns = names;
     },
+    /**
+     * Return back text/html and application/vnd.dataresource+json representations of the DataFrame
+     * @returns Media bundle / mimetype keys for Jupyter frontends
+     */
+    [Symbol.for("Jupyter.display")]() {
+      // Implementation of the Tabular Data Resource Schema. Also used for text/html representation
+      type DataResourceField = {
+        name: string;
+        type: string;
+      };
+
+      const fields = this.getColumns().reduce(
+        (acc: DataResourceField[], curr) => {
+          acc.push({
+            name: curr.name,
+            type: mapPolarsTypeToJSONSchema(curr.dtype),
+          });
+          return acc;
+        },
+        [],
+      );
+
+      // Convert DataFrame data to row-oriented JSON
+      //
+      // TODO: Determine how to get the polars format max rows
+      //       Since pl.setTblRows just sets env var POLARS_FMT_MAX_ROWS,
+      //       we probably just have to pick a number for now.
+      //
+      const data = df.head(50).toRecords();
+
+      // Enough generated for the Tabular Data Resource Schema, on to
+      // creating an HTML table
+
+      let htmlTable = "<table>";
+
+      // Add table headers
+      htmlTable += "<thead><tr>";
+      fields.forEach((field) => {
+        htmlTable += `<th>${escapeHTML(field.name)}</th>`;
+      });
+      htmlTable += "</tr></thead>";
+
+      // Add table data
+      htmlTable += "<tbody>";
+      df.head(10)
```

Comment:
why do we get the head of `50,` then later get the head of `10` here? `toRecords` is one of the most expensive operations, we probably shouldn't call it twice unless we really need to.

Reply:
Oops. I should be using the already created records from above. For the HTML side I kept the table size smaller.

Should I rely on `POLARS_FMT_MAX_ROWS` for all of these?

Reply:
We should check for `POLARS_FMT_MAX_ROWS` and `POLARS_FMT_MAX_COLS` but have a large max (10K rows) so it does not blow up the browser.
I usually use `df.limit(...)` or `df.head(..)` so that should work.

Reply:
I now realize there's a bit of conflicting advice in here, depending on how I implement this.

If I break up `toHTML` and `toDataResource` to individual functions then they'll need to both call `toRecords`. If I do it all in one display function, it can create the records just once. This could be a separate function that accepts the records too.

I will of course admit that it would be better for this to be a column-oriented emission but these are the protocols we have today until I can push for more Arrow everywhere.

Reply:
To elaborate, I think it'd be cool to make the `toHTML()` function behave similarly to `toRecords()` _(with no limit/head on it)_. Then use that inside of here. It should make this much more concise while also providing reusability via `toHTML()` 

```ts
[jupyterDisplay]() {
  const html = this.limit(50).toHTML(); 
  const fields = ...
  return {
    "application/vnd.dataresource+json": { data, schema: { fields } },
    "text/html": htmlTable,
  };
}

Reply:
Ok, cool will do.